### DISCUSSION TITLE
UnoCSS integration with Tsup

### DISCUSSION THREAD

**Discussion by RBilly** [unocss/unocss#2569] [Q&A] [answered]
Hi !

I fell in love with UnoCSS and I started to work on creating a component library for a design system.
I would like to build it using tsup. Unfortunately I did not find a suitable integration.
Is there any plan to create an esbuild plugin ? 
Or is there already an existing way to do that without mixing vite and tsup, using the CLI or the runtime bundler ?

Thanks for the help !

**Comment by schelmo** [comment]
maybe call unocss in [onSuccess](https://tsup.egoist.dev/#onsuccess) of tsup?

**Reply by RBilly** [reply]
It could be worth trying. I can see it working for "dev" mode but for production build I'm not so sure.
Currently I am just using unocss cli to generate the uno.css file and import it in my src/index.ts so that tsup bundles it as well.
So my dev script looks like "unocss 'src/**/*.tsx' --watch && tsup 'src/index.ts' --watch" and my build script is the same without the watch.
It's working but I think the integration could be better. Using onSuccess may be a step toward that thanks

**Reply by zyyv** [reply, marked as answer]
FYI https://github.com/onu-ui/onu-ui/blob/17dbe78eac51198428b8351435a4ef6dafa4ac3b/packages/preset/package.json#L37

**Reply by RBilly** [reply]
Thanks I did not thought about looking how it was used in Onu-UI lib (great lib btw)
So indeed moving the css generation of unocss to the consumer of the lib rather than trying bundle the generated css in the lib works great for me as well, so I did that thanks :)
I still have a minor issue. At the moment the consumer of my lib is Storybook. And now that storybook is configured with unocss I got this error: 
[unocss] entry module not found, have you add `import 'uno.css'` in your main entry?
It's non blocking and if I discard it I can still use storybook preview and see my styled component just fine.
But do you get any idea why ? Is it something specific to Storybook or did I missed something in the setup (I use vite)  ?
(I found this old issue https://github.com/unocss/unocss/issues/138 which is similar but seems to be solved)

**Reply by zyyv** [reply]
> [unocss] entry module not found, have you add import 'uno.css' in your main entry

Yes, you must add `import 'uno.css' firstly

**Reply by RBilly** [reply]
My mistake was very dumb, I added the UnoCSS() to the vite plugins using the `async viteFinal(config)` in the StorybookConfig.
By using the vite.config.ts file instead, I do not have the issue.
Thanks for the help
